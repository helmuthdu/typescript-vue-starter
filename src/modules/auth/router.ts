import SignInPage from '@/modules/auth/routes/sign-in/sign-in.page.vue';

export const AUTHENTICATION_ROUTES = Object.freeze({
  SIGN_IN: 'auth-sign-in-route'
});

export const authRoutes = [
  {
    path: '/sign-in',
    component: () => import('@/layout/empty.layout.vue'),
    children: [{ path: '', name: AUTHENTICATION_ROUTES.SIGN_IN, component: SignInPage }]
  }
];

export default authRoutes;
