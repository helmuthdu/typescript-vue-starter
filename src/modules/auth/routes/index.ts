export const AUTHENTICATION_ROUTES = Object.freeze({
  SIGN_IN: 'auth-sign-in-route'
});

export const routes = [
  {
    path: '/sign-in',
    component: () => import('../layouts/empty.layout.vue'),
    children: [
      {
        path: '',
        name: AUTHENTICATION_ROUTES.SIGN_IN,
        component: () => import('./sign-in/sign-in.route.vue')
      }
    ]
  }
];
