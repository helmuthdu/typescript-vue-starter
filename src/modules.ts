import * as authModule from '@/modules/auth';

export type State = authModule.State;
export const stores = [authModule.stores];
export const routes = [authModule.routes];
