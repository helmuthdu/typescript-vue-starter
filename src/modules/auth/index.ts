import { routes } from './routes';
import * as stores from './stores';

export type State = stores.State;

export { routes, stores };
