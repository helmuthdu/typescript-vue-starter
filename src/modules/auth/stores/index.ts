import * as auth from './modules/auth';

export type State = Readonly<{
  [auth.name]: auth.State;
}>;

export { auth };
