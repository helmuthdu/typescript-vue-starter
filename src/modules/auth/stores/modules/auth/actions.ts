import { State as AppState } from '../../';
import { authApi, AuthenticatePayload } from '../../../api/auth.api';
import { ActionContext, ActionTree } from 'vuex';
import { State } from './state';
import { AuthActions } from './types';

export interface Actions<S, R> extends ActionTree<S, R> {
  [AuthActions.LOGIN]: (context: ActionContext<S, R>, payload: any) => void;
  [AuthActions.LOGOUT]: (context: ActionContext<S, R>) => void;
}

export const actions: Actions<State, AppState> = {
  async [AuthActions.LOGIN]({ commit }, payload: AuthenticatePayload) {
    commit(AuthActions.SET_USER, {
      ...(await authApi.get(payload)).data,
      isLogged: true
    });
  },
  [AuthActions.LOGOUT]({ commit }) {
    commit(AuthActions.SET_USER, { username: '', email: '', isLogged: false });
  }
};
