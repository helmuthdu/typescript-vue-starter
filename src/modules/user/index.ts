import { routes } from './routes';
import * as stores from './stores';

export type State = Readonly<{
  [stores.user.name]: stores.user.State;
}>;

export { routes, stores };
