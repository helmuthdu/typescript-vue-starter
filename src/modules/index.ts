import * as rootModule from './root';
import * as userModule from './user';

export type State = userModule.State;

export const routes = [rootModule.routes, userModule.routes];
export const stores = [Object.values(userModule.stores)];

export default {
  routes,
  stores
};
