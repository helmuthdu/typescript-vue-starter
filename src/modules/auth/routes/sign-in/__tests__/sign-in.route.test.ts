import SignInRoute from '@/modules/auth/routes/sign-in/sign-in.route.vue';
import { AuthActions, State } from '../../../stores/modules/auth';
import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vuex, { Store } from 'vuex';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(Vuetify);

describe('Auth/Route -> SignIn', () => {
  let store: Store<State>;
  let wrapper: Wrapper<SignInRoute>;

  beforeEach(() => {
    store = new Vuex.Store({
      actions: {
        [AuthActions.LOGIN]: jest.fn()
      }
    });
    wrapper = shallowMount(SignInRoute, {
      store,
      localVue
    });
  });

  it('should renders with props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
