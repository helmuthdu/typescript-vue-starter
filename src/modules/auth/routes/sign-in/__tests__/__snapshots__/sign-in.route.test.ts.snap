// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeRoute component should renders with props 1`] = `
<div>
  <form><input label="Email" required="required"> <input type="password" label="Password" required="required"> <button color="pink" dark="">Sign-in</button></form>
</div>
`;
